### ISSUE TITLE
Watch for updated terms & warn user should be high priority

### ISSUE THREAD

**Issue by toyhaus** [tosdr/phoenix#173]
TOS and privacy policies change, unfortunately, so I want to make a strong plea for keeping track of the date of every Phoenix-project assessment of a TOS--and showing a warning like "OUTDATED" when a TOS has been updated more recently than the assessment/grade on Phoenix. Also you might link to paranoidpaul.com terms/privacy policy change pages. That way, users can see exactly what changed, immediately, even before the new terms have been reassessed and graded yet. (Maybe you could watch for changes at paranoidpaul.com as a way to trigger the "outdated" warning even.)

Keep up the good work!

**Comment by piks3l** [comment]
Hey ! Thanks for the feedback !

Actually, we already have one project (with two iterations) that does it : [TosBack 2](https://github.com/tosdr/tosback2) and its little sibling, [ToSBack3](https://github.com/tosdr/ToSBack3). In short, they are scrappers that check when ToS change on website. Those projects are pretty complicated and need better expertise to update them.

When Phoenix will run with a good beta or production version (in the coming 6 months or year), I should look at this. 

Nonetheless, we could implement alerts on ToS on Phoenix when a service's class changes. This sounds more feasible.

Hope it helps :)

**Comment by michielbdejong** [comment]
Yes, the website for that scraper service is https://tosback.org .

**Comment by hugoroy** [comment]
Does the system records when a contribution was made?

This way we could in the future have a flag when the Tosback date is more recent than the contribution date, and let users click a button to say that they have checked wither a point is still up to date or not.

**Comment by michielbdejong** [comment]
I was thinking we should make points refer to a specific paragraph, but even without that, using the submit date we can also mark points about services whose terms have changed since, good idea!

**Comment by michielbdejong** [comment]
Working on the link between ToS;DR points and Tosback2 crawls now. I'll add a 'quote' to each point. On Wednesday, I discussed this with @Vinnl and we concluded that it would neat to store the following fields on a point, to uniquely identify a quote:

* document
* revision
* line number start
* column start
* line number end
* column end

Although I agree that would be enough to identify for instance [lines 24 and 25 of 500px's privacy policy, as crawled on 16 November 2012](https://github.com/tosdr/tosback2/blob/abdca58/crawl/500px.com/Privacy%20Policy.txt#L24-L25), I now do think it would be useful to also store the actual text snippet:

> 500px and its authorized agents automatically receive and record certain "traffic data" on their server logs from your browser including your IP address, 500px cookie information, and the page you requested. 500px uses this traffic data to help diagnose problems with its servers, analyze trends and administer the website.

The reason I want to store this in the database, is because it would enable the use of [Postgres Trigram matching](https://www.postgresql.org/docs/9.6/static/pgtrgm.html) for suggesting cases. This could be a big help in solving #425.
